src/Models/HydrostaticFreeSurfaceModels/hydrostatic_free_surface_model.jl

```diff
@@ -0,0 +1,160 @@
+        advection :: V         # Advection scheme for tracers
+         buoyancy :: B         # Set of parameters for buoyancy model
+         coriolis :: R         # Set of parameters for the background rotation rate of `Model`
+     free_surface :: S         # Free surface parameters and fields
+          forcing :: F         # Container for forcing functions defined by the user
+          closure :: E         # Diffusive 'turbulence closure' for all model fields
+        particles :: P         # Particle set for Lagrangian tracking
+       velocities :: U         # Container for velocity fields `u`, `v`, and `w`
+          tracers :: C         # Container for tracer fields
+         pressure :: Φ         # Container for hydrostatic pressure
+    diffusivities :: K         # Container for turbulent diffusivities
+      timestepper :: TS        # Object containing timestepper fields and parameters
+end
+
+"""
+    HydrostaticFreeSurfaceModel(;
+                   grid,
+           architecture = CPU(),
+                  clock = Clock{eltype(grid)}(0, 0, 1),
+              advection = CenteredSecondOrder(),
+               buoyancy = SeawaterBuoyancy(eltype(grid)),
+               coriolis = nothing,
+                forcing = NamedTuple(),
+                closure = IsotropicDiffusivity(eltype(grid), ν=ν₀, κ=κ₀),
+    boundary_conditions = NamedTuple(),
+                tracers = (:T, :S),
+              particles = nothing,
+             velocities = nothing,
+               pressure = nothing,
+          diffusivities = nothing,
+    )
+
+Construct an hydrostatic `Oceananigans.jl` model with a free surface on `grid`.
+
+Keyword arguments
+=================
+
+    - `grid`: (required) The resolution and discrete geometry on which `model` is solved.
+    - `architecture`: `CPU()` or `GPU()`. The computer architecture used to time-step `model`.
+    - `gravitational_acceleration`: The gravitational acceleration applied to the free surface
+    - `advection`: The scheme that advects velocities and tracers. See `Oceananigans.Advection`.
+    - `buoyancy`: The buoyancy model. See `Oceananigans.Buoyancy`.
+    - `closure`: The turbulence closure for `model`. See `Oceananigans.TurbulenceClosures`.
+    - `coriolis`: Parameters for the background rotation rate of the model.
+    - `forcing`: `NamedTuple` of user-defined forcing functions that contribute to solution tendencies.
+    - `boundary_conditions`: `NamedTuple` containing field boundary conditions.
+    - `tracers`: A tuple of symbols defining the names of the modeled tracers, or a `NamedTuple` of
+                 preallocated `CenterField`s.
+"""
+function HydrostaticFreeSurfaceModel(; grid,
+                architecture::AbstractArchitecture = CPU(),
+                                             clock = Clock{eltype(grid)}(0, 0, 1),
+                                         advection = (momentum=VectorInvariant(), tracers=CenteredSecondOrder()),
```

Comment:
💪 

Out of curiousity I thought `VectorInvariant` changes the form of the equations being solved, but can we think of it as a different advection scheme?

Reply:
I'm generalizing the kwarg `advection` to mean something like "advection term" (both continuous equation aspects and numerical aspects).

We could try to distinguish between continuous aspects (whether we use a vector invariance form, an "interior" form, or a conservation form) and numerical aspects (numerical approximations / stencils used to implement that form).

We could also have a more hierarchical interface; something like

```julia
advection = (momentum=VectorInvariantForm(), tracers=ConservationForm(scheme=CenteredSecondOrder()))
```

I'm open to suggestion; I just did something lazy and simple here.

Reply:
Oh yeah no I think what you did is great. Just wanted to understand. Makes sense that it encapsulates both the continuous and numerical aspects.